src/Adapter/Composer.php

```diff
@@ -219,4 +219,61 @@ public function addAuthenticationToken(string $url, string $token): void
             $token
         );
     }
+
+    public function addGitlabOauthAuthentication(string $token, string $url = 'gitlab.com'): void
+    {
+        $this->command(
+            'composer',
+            'config',
+            '--auth',
+            sprintf('gitlab-oauth.%s', $url),
+            'token',
+            $token
+        );
+    }
+
+    public function addGitlabTokenAuthentication(string $token, string $url = 'gitlab.com'): void
+    {
+        $this->command(
+            'composer',
+            'config',
+            '--auth',
+            sprintf('gitlab-token.%s', $url),
+            $token
+        );
+    }
+
+    public function addGithubOauthAuthentication(string $token): void
+    {
+        $this->command(
+            'composer',
+            'config',
+            '--auth',
+            'github-oauth.github.com',
+            $token
+        );
+    }
```

Comment:
On a besoin de spécifier l'URL, sinon on ne peut pas être compatible avec Github Enterprise

```suggestion
    public function addGithubOauthAuthentication(string $token, string $url = 'github.com'): void
    {
        $this->command(
            'composer',
            'config',
            '--auth',
            sprintf('github-oauth.%s', $url),
            $token
        );
    }
```